force-app/main/default/lwc/deleteSelectedContacts/deleteSelectedContacts.js

```diff
@@ -4,66 +4,117 @@ import start from '@salesforce/apex/ContactController.start';
+                        type: "Contact"
+                    }                
+                };
+                this.allContactsMap.set(contact.Id+"", temp);
+            });
+            this.data = preparedContacts;
+            let numberOfPages = Math.ceil(this.data.length / this.perpage);
+            for (let i = 0; i < numberOfPages; i++) {
+                this.pages.push(i);
+                let startIndex = ((i+1)*this.perpage) - this.perpage;
+                let endIndex = ((i+1)*this.perpage);
+                let tmp = this.data.slice(startIndex, endIndex);
+                this.arrayDataPerPage.push(tmp);
+            }
+            this.pageData();
         }else if (result.error){
-            this.error = result.error;
+            this.error = result.error;d
             this.data = undefined;
         }
     }
 
     getSelectedRecords(event) {
-
         const selectedRows = event.detail.selectedRows;
-        this.recordsCount = event.detail.selectedRows.length;
-        let contactIdSet = new Set();
-        for (let i = 0; i < selectedRows.length; i++) {
-            contactIdSet.add(selectedRows[i].Id);            
-        }
-        if(contactIdSet){
-            this.selectedRecords = Array.from(contactIdSet);
+        if(this.selection[this.data] === undefined || selectedRows.length != this.selection[this.data].length){
+            let contactSetId = new Set();
+            for (let i = 0; i < selectedRows.length; i++) {
+            contactSetId.add(selectedRows[i].Id);           
+            }
+        this.selection[this.page] = Array.from(contactSetId);
+        this.recordsCount = this.selection.flat().length;
+        }                     
+    }
+
+    getFinalSelectedContacts(){
+        let contactsId = this.selection.flat();
+        let finalSelectedContacts = [];
+        if(contactsId.length === 0){
+            this.error = new Error("No one contacts are selected!");
+             this.dispatchEvent(
+             new ShowToastEvent({
+                    title: 'Error!!', 
+                    message: this.error.message, 
+                    variant: 'error'
+                }),
+           );           
+           throw this.error;
+        }else{
+            for (let i = 0; i < contactsId.length; i++) {
+                finalSelectedContacts.push(this.allContactsMap.get(contactsId[i])); 
+            }
         }
-        window.console.log('selectedRecords ====> ' + this.selectedRecords);
+        return finalSelectedContacts;
     }
 
-    deleteAll() {
-        start({source: this.selectedRecords})
+    deleteAll() {        
+        start({frontSource: JSON.stringify(this.getFinalSelectedContacts())})
         .then(result => {
-            window.console.log('result ====> ' + result.body);
+            window.console.log('result ====> ' + result);
             this.dispatchEvent(
                 new ShowToastEvent({
                     title: 'Success!!', 
```

Comment:
one exclamation point is more than enough

Reply:
Fixed